copy/copy.go

```diff
@@ -92,6 +93,7 @@ type Options struct {
 	ReportWriter     io.Writer
 	SourceCtx        *types.SystemContext
 	DestinationCtx   *types.SystemContext
+	LayerCopyHook    func(types.BlobInfo) bool
```

Comment:
we pass the layer id so really you could do anything you want, including keeping a table of images you want to keep, which is exactly what box does.